### PR TITLE
drift fit improvements

### PR DESCRIPTION
**Pull request by brownd1978**
3 updates:
-Improved StrawResponse drift calibration, with separate signed and unsigned drift error estimation 
-SOFIE inference and TensorFlow training for KinKal track fit ANNs
-Unsigned drift radius used to estimate null hit error

### CODE REVIEW

**Comment by kutschke** [comment]
@bonventre a gentle reminder that this review is in your input queue

**Comment by brownd1978** [comment]
@FNALbuild run build test

**Comment by kutschke** [comment]
@bonventre a gentle reminder that this review is in your input queue

**Comment by kutschke** [comment]
@FNALbuild run build test

**Review by bonventre — CHANGES_REQUESTED** [review]

**Comment by brownd1978** [comment]
@FNALbuild run build test

File: TrackerConditions/src/StrawResponse.cc [hunk 1]
```diff
@@ -85,21 +111,21 @@ namespace mu2e {
...
+    dinfo.LorentzAngle_ = phi;
+    dinfo.cDrift_ = _strawDrift->T2D(dtime,phi,false); // allow values outside the physical range at this point
+    int halfrange(2); // should be a parameter TODO
+    double dcorr, dcorrslope;
+    interpolateCalib(_driftOffBins,_driftOffset, dinfo.cDrift_, halfrange, dcorr, dcorrslope);
+    dinfo.rDrift_ = dinfo.cDrift_ -dcorr;
+    // note 'Velocity' is really dR/dt (change in calibrated drift distance WRT measured time), not a true physical velocity
+    dinfo.driftVelocity_ = _strawDrift->GetInstantSpeedFromD(dinfo.cDrift_)*(1.0 - dcorrslope)*_dRdTScale;
+    double serrslope,uerrslope;
+    interpolateCalib(_driftOffBins,_signedDriftRMS, dinfo.rDrift_, halfrange, dinfo.signedDriftError_, serrslope);
```
[the remark this answers is not in the document]
Comment (brownd1978) [reply TrackerConditions/src/StrawResponse.cc]: Yes, good spot.  Fixed

**Comment by brownd1978** [comment]
@FNALbuild run build test

**Comment by brownd1978** [comment]
A bunch of my test jobs from yesterday failed, please hold off merging till I can trace it back and fix the problem.

**Comment by kutschke** [comment]
> A bunch of my test jobs from yesterday failed, please hold off merging till I can trace it back and fix the problem.

Will do.

**Comment by brownd1978** [comment]
@FNALbuild run build test

**Review by bonventre — APPROVED** [review]

**Pull request merged**